packages/core-flows/src/region/steps/upsert-and-replace-region-payment-providers.ts

```diff
@@ -0,0 +1,284 @@
+import {
+  ModuleRegistrationName,
+  Modules,
+  RemoteLink,
+} from "@medusajs/modules-sdk"
+import {
+  FilterableRegionProps,
+  IPaymentModuleService,
+  IRegionModuleService,
+  RemoteQueryFunction,
+} from "@medusajs/types"
+import { createStep, StepResponse } from "@medusajs/workflows-sdk"
+import {
+  arrayDifference,
+  ContainerRegistrationKeys,
+  MedusaError,
+  promiseAll,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
+
+export interface UpsertAndReplaceRegionPaymentProvidersStepInput {
+  input:
```

Comment:
suggestion: `input` doesn't need to be an array

To update multiple regions with a set of payment providers, you would do:
```
upsertAndReplaceRegionPaymentProvidersStep({
  regions: { currency_code: "dkk" },
  payment_providers: ["mobilepay", "stripe"]
})
```
Here, we are updating all regions with a specific currency code.

We don't have to support updating many region with many selectors, in my opinion. I would rather optimize for simplifying the step consumption here.